src/registrar/assets/js/get-gov-admin.js

```diff
@@ -898,30 +938,43 @@ document.addEventListener('DOMContentLoaded', function() {
             // Update the "contact details" blurb beneath senior official
             updateContactInfo(data);
             showElement(contactList.parentElement);
-
+            
+            // Get the associated senior official with this federal agency
             let seniorOfficialId = data.id;
             let seniorOfficialName = [data.first_name, data.last_name].join(" ");
-            if (!seniorOfficialId || !seniorOfficialName || !seniorOfficialName.trim()){
-                // Clear the field if the SO doesn't exist
-                $seniorOfficial.val("").trigger("change");
-                return;
-            }
-
-            // Add the senior official to the dropdown.
-            // This format supports select2 - if we decide to convert this field in the future.
-            if ($seniorOfficial.find(`option[value='${seniorOfficialId}']`).length) {
-                // Select the value that is associated with the current Senior Official.
-                $seniorOfficial.val(seniorOfficialId).trigger("change");
-            } else { 
-                // Create a DOM Option that matches the desired Senior Official. Then append it and select it.
-                let userOption = new Option(seniorOfficialName, seniorOfficialId, true, true);
-                $seniorOfficial.append(userOption).trigger("change");
+            if ($seniorOfficial && $seniorOfficial.length > 0) {
+                // If the senior official is a dropdown field, edit that
+                updateSeniorOfficialDropdown($seniorOfficial, seniorOfficialId, seniorOfficialName);
+            }else {
+                if (readonlySeniorOfficial) {
+                    let seniorOfficialLink = `<a href=/admin/registrar/seniorofficial/${seniorOfficialId}/change/>${seniorOfficialName}</a>`
+                    readonlySeniorOfficial.innerHTML = seniorOfficialName ? seniorOfficialLink : "-";
+                }
             }
         })
         .catch(error => console.error("Error fetching senior official: ", error));
 
     }
 
+    function updateSeniorOfficialDropdown(dropdown, seniorOfficialId, seniorOfficialName) {
+        if (!seniorOfficialId || !seniorOfficialName || !seniorOfficialName.trim()){
+            // Clear the field if the SO doesn't exist
```

Comment:
this might be a nitpick. Would `!seniorOfficialName` and `!seniorOfficialName.trim()` both pass if `seniorOfficialName `was invalid?